drracket/drracket/private/language.rkt

```diff
@@ -350,27 +348,25 @@
   ;; simple-module-based-language-render-value/format : TST settings port (union #f (snip% -> void)) (union 'infinity number) -> void
   (define (simple-module-based-language-render-value/format value settings port width)
     (let-values ([(converted-value write?)
-                  (call-with-values
-                      (lambda ()
-                        (simple-module-based-language-convert-value value settings))
-                    (case-lambda
-                     [(converted-value) (values converted-value #t)]
-                     [(converted-value write?) (values converted-value write?)]))])
-      (let ([pretty-out (if write? pretty-write pretty-print)])
-        (setup-printing-parameters
-         (λ ()
-            (cond
-             [(simple-settings-insert-newlines settings)
-              (if (number? width)
-                  (parameterize ([pretty-print-columns width])
-                    (pretty-out converted-value port))
-                  (pretty-out converted-value port))]
-             [else
-              (parameterize ([pretty-print-columns 'infinity])
-                (pretty-out converted-value port))
-              (newline port)]))
-         settings
-         width))))
+                  (call-with-values (lambda ()
+                                      (simple-module-based-language-convert-value value settings))
```

Comment:
I'm not sure why this one is a good idea (the removal of this newline in particular)

Reply:
It's a side effect of running `fmt` on the surrounding definition context form. Resyntax operates on definition contexts by generating a replacement for the entire context form, even if the only change it wants to make is to a single subform of the context. Then, Resyntax formats the entire replacement form, resulting in seemingly pointless formatting changes near the code Resyntax actually wants to change. There are two possible fixes:

1. Make `fmt` better. I'm not sure why `fmt` thought indenting this code so much was an improvement. Perhaps @sorawee has thoughts here.
2. Make Resyntax format just the subform(s) in a definition context replacement that changed. I think this is possible, though tricky. I've filed jackfirth/resyntax#282 to address this.

Reply:
Algorithmically, the code fits within the width limit (102 characters IIRC), and it has fewer number of lines than the other format that enters a newline, so the output layout is the best from the algorithm's point of view.

IMO, `call-with-values` is the borderline case. I don't think it's too long to makes this layout look "bad", but I can see why someone might not like this. And longer identifiers would look bad to me as well (e.g. `call-with-parameterization`).

Possible fixes include:
- Reduce the width limit
- Add a rule so that if the "function" position has more than X characters, the following layout won't be used.

```
(func arg1
      arg2
      arg3)
```

And then we need to come up with this X value that people find pleasing.

Reply:
In general, I think lines with very large amounts of leading indentation should be avoided even if doing so results in more newlines in the code. They're just too unreadable. Is that something `fmt` could take into consideration?

Reply:
Hypothetically, we could heavily penalize indentation spaces more in `fmt`. But doing so would mean:

```
(func arg1
      arg2
      arg3)
```

is inferior than:

```
(func 
 arg1
 arg2
 arg3)
```

and I hope everyone agrees that in this specific case, the former looks better.

So it comes down to: what's the precise definition for "very large amounts of leading indentation"? 

One possibility is to determine the X value that I wrote in the above post. Is there any other way?

But no, currently `fmt` doesn't take large indentation into account (large indentation usually results in going over the width limit, so that's why it usually does not occur).

Reply:
Is there any chance that fmt (or something in this process) could take edit information into account? Like "the indentation was like this before, I changed this part; reindent the new parts but keep the old parts as they were?"

... but maybe this is against the whole point of automatic formatting tools?

In that case, probably best if I just take what is offered!

Reply:
`fmt` is an _opinionated_ code formatter. It mostly doesn't care about existing format in the code (there are some exceptions). Some other code formatters are _unopinionated_, so they will try to preserve e.g. existing line breaks.

Being opinionated is fmt's primary goal. I wanted to format poorly written code that I saw in Discord, and this task includes adding missing newlines and removing excessive newlines.

------

@jackfirth what do you think about manually fixing these formats for now, while waiting for fmt improvement?

Reply:
@sorawee , okay that makes sense. I'm happy to take whatever the general algorithm decides is best and happy that there's a desire to improve cases when it makes sense!

Reply:
@rfindler Taking edit information into account would make Resyntax's job a lot harder too. Currently when Resyntax needs to write new code, it just spits out the most naive newline-free code possible and relies on `fmt` to turn it into something reasonable. If `fmt` was less opinionated and more willing to keep the code unchanged, that would probably result in some ugliness. So `fmt` being opinionated is a very valuable feature from Resyntax's perspective.

@sorawee Manually fixing these is good with me. And as for the `fmt` improvement, how about adding an extra linebreak when `func` is longer than... say... 15 characters? `call-with-values` is 16, so it'd catch that one.

Sidenote: I wish so badly I could write this:

```scheme
(some-function
    arg1
    arg2
    arg3)
```

But that's pretty radically different from what everyone's used to today.